src/jit/flowgraph.cpp

```diff
@@ -4857,7 +4859,8 @@ void Compiler::fgFindJumpTargets(const BYTE* codeAddr, IL_OFFSET codeSize, BYTE*
                     fgObserveInlineConstants(opcode, pushedStack, isInlining);
                 }
                 break;
-
+            case CEE_RET:
+                retBlocks++;
```

Comment:
As already mentioned, this is not enough. We need to handle tail calls as well, they count as returns.